Checkout testDeleteAdmin() method; not sure if it should be implemented at all given mock deletion might not be possible

zhannaklimanova opened the pull request:
TestAdminService class added; TestCustomerService + TestTechnicianService redundant object creation removed, AdminDto cleaned up

zhannaklimanova committed:
TestAdminService class added; TestCustomerService + TestTechnicianService redundant object creation removed, AdminDto cleaned up.

zhannaklimanova committed:
deleted random ReminderService class outside of all  packages

T-Syme commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java:
```diff
@@ -1,18 +1,537 @@
+	};
+	
+	lenient().when(adminRepo.save(any(Admin.class))).thenAnswer(returnParameterAsAnswer);
+	
```
`adminRepo.deleteByCustomerEmail(email)` should probably be mocked too

Edit: I might have been wrong about this. I don't know if it can be mocked

zhannaklimanova committed:
Deleted assertNull method 

Not needed b/c mock outputs already checks that the deleteAdmin method calls all the correct methods properly

T-Syme reviewed (approved):
Left a comment about a mock that shouldn't really have an impact. Also, I'm not sure about the changes to `testCreateTechnicianDuplicate()` and the customer version, but as long as it works...

sandyl289 commented on RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AdminService.java:
```diff
@@ -158,7 +159,7 @@ public AdminDto changeName(String email, String newName) throws Exception{
 	}
 	
 	@Transactional
```
```suggestion
```
This is causing an error. We don't need to put `@Transactional` over this method.

![image](https://user-images.githubusercontent.com/70775435/111055550-e89a4900-8444-11eb-8b42-03ae06ba7633.png)

sandyl289 replied:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae

sandyl289 commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java:
```diff
@@ -1,18 +1,536 @@
+		assertNotNull(admin);
+		assertEquals(adminName, admin.getName());
+		assertEquals(adminEmail, admin.getEmail());
+		
```
```suggestion
		assertNotNull(admin);
		assertEquals(adminName, admin.getName());
		assertEquals(adminEmail, admin.getEmail());
		assertEquals(adminPhone, admin.getPhoneNumber());
		assertEquals(adminAddress, admin.getAddress());
```

sandyl289 replied:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae

sandyl289 commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java:
```diff
@@ -1,18 +1,536 @@
+			//error should occur
+			assertEquals("Admin not found.", e.getMessage());
+		}
+	}
```
```suggestion
	@Test
	public void testChangePhoneNonExistentAdmin() {
				
		try {
			adminService.changePhone("nonExistentAdmin@nonExistentEmail.com", "somebody's phone");
			fail();
		} catch (Exception e) {
			//error should occur
			assertEquals("Admin not found.", e.getMessage());
		}
	}
```

sandyl289 replied:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae

sandyl289 commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java:
```diff
@@ -1,18 +1,536 @@
+	
+	
+	@Test
+	public void testChangeNamelNull() {
```
Not really important, but there's an extra "l" between Name & Null

```suggestion
	public void testChangeNameNull() {
```

sandyl289 replied:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae

sandyl289 commented on RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java:
```diff
@@ -1,18 +1,536 @@
+	public void testChangeNameNonExistentAdmin() {
+				
+		try {
+			adminService.changeName("No One", "Some One");
```
```suggestion
			adminService.changeName("NoOne@gmail.com", "Some One");
```

sandyl289 replied:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae

sandyl289 reviewed (changes requested):
- We should remove `@Transaction` over the `adminToDTO()` method
- I put some suggestions, you can ignore them if you want
Other than that, everything seems good 👍

T-Syme commented:
I'll implement Sandy's suggestions while I take care of [Adding a service as an admin](https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/94)

T-Syme commented:
Just a note before I forget: Javadoc should be added to the service methods in AdminService

T-Syme committed:
Implemented Admin suggestions + got rid of warnings

T-Syme committed:
Added ability to add a service from the admin controller

T-Syme commented:
I'm not sure if I should be adding a method to add a service in the admin controller as I did or if there's a better way of doing it...

sandyl289 reviewed (commented):
I'm not sure, but I think we don't need to put this in the AdminController since we already have this createService() in the ServiceController.

I'm looking at group 1's repo and it seems like they did not put the createAdvertisment() in [userController.java](https://github.com/McGill-ECSE321-Winter2020/project-group-01/blob/master/PetShelter-Backend/src/main/java/ca/mcgill/ecse321/petshelter/controller/UserController.java). They only have [createAdvertisement()](https://github.com/McGill-ECSE321-Winter2020/project-group-01/blob/22de27a9f1d9403489ffac559add113f7977bb5e/PetShelter-Backend/src/main/java/ca/mcgill/ecse321/petshelter/controller/AdvertisementController.java#L123-L147) in the AdvertisementController.java class. 


Their [use case diagram: ](https://github.com/McGill-ECSE321-Winter2020/project-group-01/wiki/Use-Case-Diagram#use-case-diagram)

![image](https://user-images.githubusercontent.com/70775435/111078958-49219880-84ce-11eb-9eb0-99acff48a5a8.png)

T-Syme commented:
> I'm not sure, but I think we don't need to put this in the AdminController

Sounds good. I'll revert the last commit I added and close the [Adding a service as an admin](https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/94) issue separately

T-Syme committed:
Revert "Added ability to add a service from the admin controller"

This reverts commit 5b41ddcdb434c3099cba08527842317e917c38b9.

zhannaklimanova commented:
@sandyl289 I'm not sure I understand why is it you removed the transactional part from adminToDto and other methods?

T-Syme commented:
@zhannaklimanova There are no transactions that occur when converting to an AdminDto (there's no interaction with the database). 
(If this isn't accurate, please correct me)

sandyl289 committed:
Merge branch 'main' into testAdminService

sandyl289 committed:
Remove change email tests

- We should not let the admin to change their email (primary key)
